src/Bicep.Core/Analyzers/Linter/LinterRule.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.CodeAction;
+using Bicep.Core.Configuration;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Bicep.Core.Analyzers.Linter
+{
+    public abstract class LinterRule : IBicepAnalyzerRule
+    {
+        public LinterRule(string code, string ruleName, string description, string docUri,
+                          bool enableForEdit = true,
+                          bool enableForCLI = true,
+                          DiagnosticLevel diagnosticLevel = DiagnosticLevel.Warning)
+        {
+
+            this.ConfigHelper = new ConfigHelper();
+            this.AnalyzerName = LinterAnalyzer.AnalyzerName;
+            this.Code = code;
+            this.RuleName = ruleName;
+            this.Description = description;
+            this.DocumentationUri = docUri;
+
+            this.EnabledForCLI = enableForCLI;
+            this.EnabledForEditing = enableForEdit;
+            this.DiagnosticLevel = diagnosticLevel;
+
+            LoadConfiguration();
+        }
+
+        private readonly ConfigHelper ConfigHelper;
+        public string AnalyzerName { get; }
+        public string Code { get; }
+        public string RuleName { get; }
+
+        public const string RuleConfigSection = "Linter:Rules";
+
+        // TODO: Decide how we want to manage configuration
```

Comment:
>TODO [](http://example.com/codeflow?start=11&length=4)

I'd also add this question to the list: How do we surface errors in linter configuration to the user? It would be cool if the issues showed up as separate diagnostics.